### ISSUE TITLE
Provide an option to use default transition when overloading transition configuration 

### ISSUE THREAD

**Issue by jdgreenberger** [react-navigation/react-navigation#3674]
This issue is a feature suggestion. The [feature guidelines](https://reactnavigation.org/docs/contributing.html#suggesting-a-feature) suggested I create a feature request in the issues. Please let me know if there is a more preferable location. 

Right now, if you want to specify a custom transition, you must overload the `transitionConfig` param. There are situations however, where one may want to conditionally display a custom transition, and use the default in other situations. Example:
```javascript
screenInterpolator: (props) => {
  if (props.scene.route.routeName === 'someView') {
    return customTransition(props);
  } else {
    return defaultScreenTransition(props);
  }
}
```
 One alternative, based on this [issue](https://github.com/react-navigation/react-navigation/issues/2493), is to reach into the SRC and pull out the default transition you need. This solution is not ideal because:
1) the `forHoritzontal` transition does not reflect the actual default transitions which use a different transition for iOS/android 
2) reaching into the src is bug prone if one ever intends to upgrade

I'm proposing `transitionConfig` should take an option to return to the default transition, or expose the default transition function as a top level library export.

**Comment by jdgreenberger** [comment]
Any update on this? I can try to submit a PR if the owners agree with the suggestion.

**Comment by brentvatne** [comment]
@ericvicenti - any thoughts here?

**Comment by ericvicenti** [comment]
I think it makes sense for us to export the defaults from the library so that people can override it conditionally like this.

**Comment by ericvicenti** [comment]
A PR would be appreciated!

**Comment by ayushmad** [comment]
I am not sure it should be part of the same PR if not I can create a new one. It would also be nice to have _StackViewStyleInterpolar_ exposed. 
My thinking was if Interpolator were exposed then consumer could pick more the interpolator needed for there screenTransition. 

**Use Case:**
Defaults really work really well for the app(its an IOS only app) I am working on except in one screen I wanted to add the `forFadeFromBottomAndroid` in IOS.
I was planning on implementing my own and saw that this already exists and wanted to use it directly.
Currently I am going down and pulling the interpolator from the library.


* I wanted to check if this makes sense or if you guys want consumers to use build there own interpolator.
- I can create a pull request if needed.



_Side Note:_
For the Example above where you only want to add transitions for specific screen. I get around it by not providing a config at all. Based on the code it always falls back to the default config.
```
transitionConfig: () => {
   if (props.scene.route.routeName === 'someView') {
        return customTransitionConfig;
   }
   return {};
}
```
Based on the way the transitionConfig is used all the properties fallback if not provided.
_I am not sure if this helps you get unblocked but thought this might work better for you._

**Comment by jdgreenberger** [comment]
@ayushmad for some reason when I tried to implement a top level transition config, I didn't have access to the props in order to check `if (props.scene.route.routeName === 'someView')` but it seems like they are available in the latest version. The above solution will solve the problem, thank you.

**Comment by ayushmad** [comment]
Glad it worked out. @ericvicenti Should I create a PR for my use case above ?
